### PR TITLE
Make footgun more obvious

### PR DESCRIPTION
**Pull request by greg0ire**
#### Summary

Partially addresses #4634

### CODE REVIEW

File: lib/Doctrine/DBAL/Cache/ResultCacheStatement.php [hunk 1]
```diff
@@ -189,6 +192,8 @@ public function fetchAll($fetchMode = null, $fetchArgument = null, $ctorArgs = n
     }
 
     /**
+     * Be warned that you will need to call this method until no rows are
+     * available for caching to happen.
```
Comment (SenseException) [inline lib/Doctrine/DBAL/Cache/ResultCacheStatement.php:196]:
```suggestion
     * available for caching to happen.
     *
```

File: lib/Doctrine/DBAL/Cache/ResultCacheStatement.php [hunk 2]
```diff
@@ -202,6 +207,8 @@ public function fetchColumn($columnIndex = 0)
...
```
Comment (SenseException) [inline lib/Doctrine/DBAL/Cache/ResultCacheStatement.php:212]:
```suggestion
     * available for caching to happen.
     *
```

File: lib/Doctrine/DBAL/Cache/ResultCacheStatement.php [hunk 3]
```diff
@@ -216,6 +223,8 @@ public function fetchNumeric()
...
```
Comment (SenseException) [inline lib/Doctrine/DBAL/Cache/ResultCacheStatement.php:229]:
```suggestion
     * available for caching to happen.
     *
```

File: lib/Doctrine/DBAL/Cache/ResultCacheStatement.php [hunk 4]
```diff
@@ -224,6 +233,8 @@ public function fetchAssociative()
...
```
Comment (SenseException) [inline lib/Doctrine/DBAL/Cache/ResultCacheStatement.php:240]:
```suggestion
     * available for caching to happen.
     *
```

**Review by SenseException — DISMISSED** [review]

**Comment by greg0ire** [comment]
@derrabus please review

**Comment by greg0ire** [comment]
Also @derrabus, shouldn't caching.rst have been updated with #4624?

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
> Also @derrabus, shouldn't caching.rst have been updated with #4624?

Indeed! I'll try to do that within the next days. Feel free to open an issue so we (I) don't forget.

**Comment by Ocramius** [comment]
All green, CI entries added. /cc @greg0ire

**Review by SenseException — APPROVED** [review]

**Pull request merged**